internal/service/kendra/index.go

```diff
@@ -0,0 +1,893 @@
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	name := d.Get("name").(string)
+
+	input := &kendra.CreateIndexInput{
+		ClientToken: aws.String(resource.UniqueId()),
+		Name:        aws.String(name),
+		RoleArn:     aws.String(d.Get("role_arn").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("edition"); ok {
+		input.Edition = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("server_side_encryption_configuration"); ok {
+		input.ServerSideEncryptionConfiguration = expandServerSideEncryptionConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("user_context_policy"); ok {
+		input.UserContextPolicy = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("user_group_resolution_configuration"); ok {
+		input.UserGroupResolutionConfiguration = expandUserGroupResolutionConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("user_token_configurations"); ok {
+		input.UserTokenConfigurations = expandUserTokenConfigurations(v.([]interface{}))
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	log.Printf("[DEBUG] Creating Kendra Index %s", input)
+	output, err := conn.CreateIndexWithContext(ctx, input)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Kendra Index (%s): %w", name, err))
+	}
+
+	if output == nil {
+		return diag.FromErr(fmt.Errorf("error creating Kendra Index (%s): empty output", name))
+	}
+
+	d.SetId(aws.StringValue(output.Id))
+
+	// waiter since the status changes from CREATING to either ACTIVE or FAILED
+	if _, err := waitIndexCreated(ctx, conn, d.Timeout(schema.TimeoutCreate), d.Id()); err != nil {
+		return diag.FromErr(fmt.Errorf("error waiting for Index (%s) creation: %w", d.Id(), err))
+	}
+
+	return resourceIndexRead(ctx, d, meta)
+}
+
+func resourceIndexRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).KendraConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	// region and accountId used to construct the ARN - not returned by API
+	region := meta.(*conns.AWSClient).Region
+	accountId := meta.(*conns.AWSClient).AccountID
+
+	id := d.Id()
+
+	resp, err := conn.DescribeIndexWithContext(ctx, &kendra.DescribeIndexInput{
+		Id: aws.String(id),
+	})
+
+	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, kendra.ErrCodeResourceNotFoundException) {
+		log.Printf("[WARN] Kendra Index (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting Kendra Index (%s): %w", d.Id(), err))
+	}
+
+	if resp == nil {
+		return diag.FromErr(fmt.Errorf("error getting Kendra Index (%s): empty response", d.Id()))
+	}
+
+	d.Set("arn", fmt.Sprintf("arn:aws:kendra:%s:%s:index/%s", region, accountId, id))
```

Comment:
Hi @GlennChia , just a drive by comment as formalize which additional resources to prioritize. For constructed ARNs like here, we typically use the pattern:

```go
arn := arn.ARN{
		Partition: meta.(*conns.AWSClient).Partition,
		Service:   "kendra",
		Region:    meta.(*conns.AWSClient).Region,
		AccountID: meta.(*conns.AWSClient).AccountID,
		Resource:  fmt.Sprintf("index/%s", d.Id()),
	}.String()

	d.Set("arn", arn)
```
That should also clear up the linter-related failure.

Reply:
Thanks @anGie44. The changes worked. Re-ran the tests cases as well

```
$ make testacc TESTARGS='-run=TestAccKendraIndex' PKG=kendra ACCTEST_PARALLELISM=5
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/kendra/... -v -count 1 -parallel 5  -run=TestAccKendraIndex -timeout 180m
=== RUN   TestAccKendraIndex_basic
=== PAUSE TestAccKendraIndex_basic
=== RUN   TestAccKendraIndex_serverSideEncryption
=== PAUSE TestAccKendraIndex_serverSideEncryption
=== RUN   TestAccKendraIndex_updateDescription
=== PAUSE TestAccKendraIndex_updateDescription
=== RUN   TestAccKendraIndex_updateName
=== PAUSE TestAccKendraIndex_updateName
=== RUN   TestAccKendraIndex_updateUserTokenJson
=== PAUSE TestAccKendraIndex_updateUserTokenJson
=== RUN   TestAccKendraIndex_updateTags
=== PAUSE TestAccKendraIndex_updateTags
=== RUN   TestAccKendraIndex_updateRoleArn
=== PAUSE TestAccKendraIndex_updateRoleArn
=== RUN   TestAccKendraIndex_disappears
=== PAUSE TestAccKendraIndex_disappears
=== CONT  TestAccKendraIndex_basic
=== CONT  TestAccKendraIndex_updateDescription
=== CONT  TestAccKendraIndex_updateUserTokenJson
=== CONT  TestAccKendraIndex_updateRoleArn
=== CONT  TestAccKendraIndex_disappears
--- PASS: TestAccKendraIndex_updateUserTokenJson (958.14s)
=== CONT  TestAccKendraIndex_updateName
--- PASS: TestAccKendraIndex_updateRoleArn (968.86s)
=== CONT  TestAccKendraIndex_serverSideEncryption
--- PASS: TestAccKendraIndex_basic (1880.87s)
=== CONT  TestAccKendraIndex_updateTags
--- PASS: TestAccKendraIndex_disappears (1897.32s)
--- PASS: TestAccKendraIndex_updateDescription (1901.10s)
--- PASS: TestAccKendraIndex_updateName (1942.58s)
--- PASS: TestAccKendraIndex_serverSideEncryption (1939.64s)
--- PASS: TestAccKendraIndex_updateTags (1032.02s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/kendra     2912.943s
```